rpc/src/rpc.rs

```diff
@@ -704,9 +704,29 @@ impl JsonRpcRequestProcessor {
                 .await?;
 
             for (partition_index, addresses) in partition_index_addresses.iter() {
-                let slot = *block_list
-                    .get(*partition_index)
-                    .ok_or_else(Error::internal_error)?;
+                let slot = *block_list.get(*partition_index).ok_or_else(|| {
+                    // If block_list.len() too short to contain
+                    // partition_index, the epoch rewards period must be
+                    // currently active.
+                    let rewards_complete_block_height = epoch_boundary_block
+                        .block_height
+                        .map(|block_height| {
+                            block_height
+                                .saturating_add(num_partitions as u64)
+                                .saturating_add(1)
+                        })
+                        .expect(
+                            "every block after partitioned_epoch_reward_enabled should have a \
+                             populated block_height",
+                        );
+                    {
```

Comment:
is this scoped on purpose?

Reply:
Shoot, no. I originally had the `EpochRewardsPeriodActive` variant only include the slot, and forgot to clean this up when I prepended in the closure for the additional values.